prov/shm/src/smr_dsa.c

```diff
@@ -184,6 +203,33 @@ static int dsa_open_wq(struct accfg_wq *wq)
 	return fd;
 }
 
+static bool dsa_write_is_supported(int fd)
+{
+	int ret;
+	int max_comp_checks = 65536;
+	struct dsa_hw_desc desc __attribute__((aligned(64))) = {0};
+	struct dsa_completion_record comp __attribute__((aligned(32))) = {0};
+
+	desc.opcode = DSA_OPCODE_NOOP;
+	desc.flags = IDXD_OP_FLAG_CRAV | IDXD_OP_FLAG_RCR;
+	desc.completion_addr = (unsigned long) &comp;
+
+	// make sure writes (e.g., comp.status = 0) are ordered wrt to write()
+	{ asm volatile("sfence":::"memory"); }
+
+	ret = write(fd, &desc, sizeof(desc));
+
+	if (ret != sizeof(desc))
+		return false;
+
+	while (comp.status == 0 && max_comp_checks > 0) {
```

Comment:
Suggest use DSA_COMP_NONE here to avoid any confusion.  0 is often equivalent to SUCC in other definitions.